### DISCUSSION TITLE
[5.x] FR: Ability to drag entries between matrix fields that supports the same entry types

### DISCUSSION THREAD

**Discussion by ZapaNOR** [craftcms/cms#14469] [Ideas] [answered]
The entrification of the matrix field has come a long way to replace the functionality plugins like NEO has provided.
One of the few major hurdles I am facing while testing at the moment is when working with multi layer matrix fields.

Having a layout like "section -> column -> image", some times you create a lot of blocks, design them like you want, and then the customer asks; "can you move this to the bottom". As things are now, if I have created a Column, there is no way to move this Column into another section, I would have to create all the content blocks and configure them anew.

I'm not sure the ramifications of this suggestion, but if I am dragging a block (entry) in a matrix field, I would like to be able to drop it in any other matrix field that also supports that block's entry type.

**Comment by adrianjean** [comment]
Have a look at the [Smith](https://plugins.craftcms.com/smith) plugin. It allows you to copy blocks using the contextual menu not only within the same Matrix field, but across sections and entries using the same field — similar to NEO's copy past feature.

**Comment by brandonkelly** [comment, marked as answer]
Craft 5.7 is adding the ability to copy/paste entries and addresses across sections/fields 🎉 (#16888)

**Reply by brandonkelly** [reply]
Craft 5.7.0 is out with copy-paste support for entries and addresses 🎉

**Reply by alexanderbuergin** [reply]
Awesome news! We’ve been using the copy-paste feature extensively over the past few months—such a time-saver.

One question though: is there a reason a drag-and-drop feature isn’t planned? Being able to move elements instead of just duplicating them would be incredibly helpful. It avoids unnecessary duplication, reduces the number of element IDs in the database, and feels like a more intuitive UX for many use cases.

In essence, we’re talking about a move function here—not just copy/delete. I’d really love to see that as an option in the future. It would massively improve usability, especially for Matrix fields with lots of blocks.
